### ISSUE TITLE
Bamboo binaries un-tar step is failing

### ISSUE THREAD

**Issue by vrmerlin** [joshbeard/puppet-bamboo#17]
I'm using the following configuration:

```
mod 'joshbeard-bamboo', '3.4.1'
mod 'puppetlabs-stdlib', '4.25.0'
mod 'puppet-staging', '2.2.0'
```

with this code:

```
class { 'bamboo':
  version                => '6.4.1',
  installdir             => '/opt/atlassian-bamboo',
  homedir                => '/our-atlassian/bamboo',
  user                   => 'bamboouser',
  uid                    => '3300',
  java_home              => '/usr/java/default',
  jvm_xmx                => '3584m',
  jvm_xms                => '3584m',
  manage_service         => false,
}
```

and getting the following error, which appears to happen after the binary is downloaded from the vendor site:

```
Error: 'tar xzf /opt/staging/bamboo/atlassian-bamboo-6.4.1.tar.gz --strip=1' returned 2 instead of one of [0]                                                                                                      an-bamboo-6.4.1.tar.gz --strip=1' returned 2
Error: /Stage[main]/Bamboo::Install/Staging::Extract[atlassian-bamboo-6.4.1.tar.gz]/Exec[extract atlassian-bamboo-6.4.1.tar.gz]/returns: change from 'notrun' to ['0'] failed: 'tar xzf /opt/staging/bamboo/atlassian-bamboo-6.4.1.tar.gz --strip=1' returned 2 instead of one of [0]
```

If I untar the package manually, it works fine, so I know the binary is ok. Why is it failing in the puppet script though?

**Comment by vrmerlin** [comment]
This is a permissions issue. Perhaps the staging:extract works differently now. In any case, the staging module is obsolete, and instead the archive module should be used. I was able to put together a fix that now works for me, in a fork. But really, we should update this version. I'd put together a pull request, if someone was maintaining the project...?

**Comment by joshbeard** [comment]
@vrmerlin I apologize for the lack of response. If you submit a pull request for it, I'd be happy to merge it in.

**Comment by joshbeard** [comment]
This may be resolved by PR #21, which switches to the archive module.

**Comment by joshbeard** [comment]
Now using the 'archive' module. Please open a new ticket if there's related issues.
